front/lib/api/assistant/actions/websearch.ts

```diff
@@ -124,25 +123,22 @@ export class WebsearchConfigurationServerRunner extends BaseActionConfigurationS
   // WebSearch shares results count across web search actions of a same step and uses citations for
   // these.
   getCitationsCount({
+    agentConfiguration,
     stepActions,
   }: {
+    agentConfiguration: AgentConfigurationType;
     stepActions: AgentActionConfigurationType[];
   }): number {
-    const actionCount = stepActions.filter(
-      (a) => a.sId === this.actionConfiguration.sId
-    ).length;
-
-    if (actionCount === 0) {
-      throw new Error("Unexpected: found 0 websearch actions");
-    }
-
-    return Math.ceil(WEBSEARCH_ACTION_NUM_RESULTS / actionCount);
+    return getCitationsCount({
```

Comment:
`nit` it feels confusing to have the same name for the method and for the function.